packages/charting/src/components/Legends/Legends.base.tsx

```diff
@@ -238,12 +241,21 @@ export class LegendsBase extends React.Component<ILegendsProps, ILegendState> {
         onBlur={onMouseOut}
         data-is-focusable={true}
       >
-        <div className={classNames.rect} />
+        <div className={this._getShapeClass(classNames, legend)} />
```

Comment:
GitHub won't let me comment on the code above but it looks like classNames here is calling our style function again? This is going to be bad for perf. We also already have this._classNames. Is this something we can fix while you're in here to make a bit more clear?

Reply:
Each button have a different color so we need to calling our style function again for each button and can't directly use this._classNames.